cpp/src/io/orc/stripe_data.cu

```diff
@@ -1789,17 +1789,29 @@ __global__ void __launch_bounds__(block_size)
                 seconds += get_gmt_offset(tz_table.ttimes, tz_table.offsets, seconds);
               }
               if (seconds < 0 && nanos != 0) { seconds -= 1; }
-              if (s->chunk.ts_clock_rate) {
-                duration_ns d_ns{nanos};
-                d_ns += duration_s{seconds};
-                static_cast<int64_t*>(data_out)[row] =
-                  d_ns.count() * s->chunk.ts_clock_rate /
-                  duration_ns::period::den;  // Output to desired clock rate
-              } else {
-                cudf::duration_s d{seconds};
-                static_cast<int64_t*>(data_out)[row] =
-                  cuda::std::chrono::duration_cast<cudf::duration_ns>(d).count() + nanos;
+
+              duration_ns d_ns{nanos};
+              d_ns += duration_s{seconds};
```

Comment:
To me it seems like we should not always represent the entire duration as `ns`. When the output clock rate is larger, the original code used `ts_clock_rate` to scale seconds directly to the desired rate (instead of nanoseconds and then output rate, as we do now). 
https://github.com/rapidsai/cudf/blob/442009bafe30e39f86a47fc451309085cce5f179/cpp/src/io/orc/stripe_data.cu#L1791-L1794
AFAICT, this avoids the overflow seen here.

Reply:
Yeah, we shouldn't be doing the sum of nanoseconds + seconds in the beginning. 


All of the logic in this `TIMESTAMP` case should be completely refactored. It's inscrutable as it is now.
- Why is `seconds` signed by `nanos` is unsigned?
- What is going on with this logic `nanos = (nanos >> 3) * kTimestampNanoScale[nanos & 7];`? 
- What is going on here? `if (seconds < 0 && nanos != 0) { seconds -= 1; }` 


This is not intended to be final, but to point you in the right direction:
```
   auto const [seconds, nanos] = [&](){
      auto const sec = duration_s{s->vals.i64[t + vals_skipped] + s->top.data.utc_epoch};
      duration_s const timezone_offset = tz_table.ttimes.empty() ? 0 : get_gmt_offset(tz_table.ttimes, tz_table.offsets, sec);
      duration_ns const nanos = (secondary_val / 8) * kTimestampNanoScale[ secondary_val % 8]; // what is going on here?
      // what's going on with subtracting 1 from seconds?
      return thrust::make_pair( (sec < 0 and nanos != 0) ? sec - 1 : sec, nanos);
   }();

   static_cast<int64_t*>(data_out)[row] = to_timepoint(s->chunk.timestamp_type_id, seconds, nanos).count();
```

Inside `to_timepoint` you cast `seconds` and `nanos` to the specified resolution and then do the sum.

Reply:
Thanks, I noticed this issue also thus casting timestamps to the desired precision first then doing the sum and tests still failed. I also tried to revert all my chrono changes back to integer operations in orc but this didn't solve the problem either. Reverting my whole chrono PR to see if I can find more clues...